src/utils.jl

```diff
@@ -1,4 +1,10 @@
-using Printf
+# Adapt an offset CuArray to work nicely with CUDA kernels.
+Adapt.adapt_structure(to, x::OffsetArray) = OffsetArray(adapt(to, parent(x)), x.offsets)
+
+# Need to adapt SubArray indices as well.
+# See: https://github.com/JuliaGPU/Adapt.jl/issues/16
+#Adapt.adapt_structure(to, A::SubArray{<:Any,<:Any,AT}) where {AT} =
+#    SubArray(adapt(to, parent(A)), adapt.(Ref(to), parentindices(A)))

```

Comment:
Why was this commented out?

Reply:
Hmm... I wanted to test if it was needed. GPU tests pass without it, but perhaps it's necessary for something? Feel free to uncomment if so.

Reply:
Trying to remember why I needed it. Perhaps it was something in a script.